src/stepfunctions/steps/sagemaker.py

```diff
@@ -473,13 +576,19 @@ def __init__(self, state_id, tuner, job_name, data, wait_for_completion=True, ta
         super(TuningStep, self).__init__(state_id, **kwargs)
 
 
-class ProcessingStep(Task):
+class ProcessingStep(SageMakerTask):
```

Comment:
breaking change??

the class [constructor]((https://aws-step-functions-data-science-sdk.readthedocs.io/en/stable/sagemaker.html?highlight=ProcessingStep#stepfunctions.steps.sagemaker.ProcessingStep)) used to take in a Task but now that's been changed. when customers upgrade versions, won't their existing code fail?

we cannot make breaking changes as we need to follow semantic versioning while releasing minor version updates.

Reply:
This changes ProcessingStep's base class, but not the constructor arguments. With this change, instead of calling Task's constructor directly in __init__(), we call SageMakerTask's constructor which in turn calls Task's constructor. 

Before: 
- ProcessingtStep.__init__()
   - Task.__init__()

After:
- ProcessingtStep.__init__()
   - SageMakerTask.__init__()
      - Task.__init__()